File: 0300. Longest Increasing Subsequence/dp.scala
object Solution {
    def lengthOfLIS(v: Array[Int]): Int = {
        var len = v.length;
        var ret = 0;
        var dp: Array[Int] = Array.fill[Int](len + 1)(1);
        for(i <- 1 until len + 1){
            for(j <- 1 until i){
                if(v(i - 1) > v(j - 1)){
                    dp(i) = math.max(dp(j) + 1, dp(i));
                }
            }
            ret = math.max(dp(i), ret);
        }
        ret
    }
}

File: offer/JZ42 连续子数组的最大和/dp.java
public class Solution {
    public int FindGreatestSumOfSubArray(int[] v) {
        if(v.length == 0)
            return 0;
        int len = v.length, ret = v[0];
        int[] dp = new int[len];
        dp[0] = v[0];
        for(int i = 1; i < len; ++i){
            dp[i] = Math.max(v[i], dp[i - 1] + v[i]);
            ret = Math.max(ret, dp[i]);
        }
        return ret;
    }
}